admin-dev/themes/default/template/controllers/cart_rules/product_rule_itemlist.tpl

```diff
@@ -56,3 +57,24 @@
 	$('#product_rule_select_{$product_rule_group_id|intval}_{$product_rule_id|intval}_add').on('click', function() { addCartRuleOption(this); updateProductRuleShortDescription(this); });
 	$(function() { updateProductRuleShortDescription($('#product_rule_select_{$product_rule_group_id|intval}_{$product_rule_id|intval}_add')); });
 </script>
+<script>
+
+	var input = document.getElementById("search_unselected");
+	var select = document.getElementById("product_rule_select_{$product_rule_group_id|intval}_{$product_rule_id|intval}_1");
+	input.addEventListener("input", function() {
+
+		var filter = input.value.toUpperCase();
+		var options = select.getElementsByTagName("option");
+		for (var i = 0; i < options.length; i++) {
+
+			var text = options[i].textContent || options[i].innerText;
+			text = text.toUpperCase();
+
+			if (text.indexOf(filter) > -1) {
+				options[i].style.display = "";
+			} else {
+				options[i].style.display = "none";
+			}
```

Comment:
This can be improve with a ternary I think?
```suggestion
				options[i].style.display = (text.indexOf(filter) > -1) ? '' : 'none';
```

Reply:
Ok thanks , works ok